deepview_profile/commands/analysis.py

```diff
@@ -0,0 +1,161 @@
+        help="The complete analysis of all methods"
+    )
+    parser.add_argument(
+        "-breakdown", "--measure-breakdown",
+        action="store_true",
+        help="Adds breakdown data to results"
+    )
+    parser.add_argument(
+        "-throughput", "--measure-throughput",
+        action="store_true",
+        help="Adds throughput data to results"
+    )
+    parser.add_argument(
+        "-predict", "--habitat-predict",
+        action="store_true",
+        help="Adds habitat data prediction to results"
+    )
+    parser.add_argument(
+        "-utilization", "--measure-utilization",
+        action="store_true",
+        help="Adds utilization data to results"
+    )
+    parser.add_argument(
+        "-energy", "--energy-compute",
+        action="store_true",
+        help="Adds energy use to results"
+    )
+    parser.add_argument(
+        "-o", "--output",
+        help="The location where the complete report should be stored",
+        required=True
+    )
+    parser.add_argument(
+        "--log-file",
+        help="The location of the log file",
+    )
+    parser.add_argument("--debug", action="store_true", help="Log debug messages.")
+    parser.set_defaults(func=main)
+
+
+
+def measure_breakdown(session, nvml):
+    print("analysis: running measure_breakdown()")
+    yield session.measure_breakdown(nvml)
+    release_memory()
+
+def measure_throughput(session):
+    print("analysis: running measure_throughput()")
+    yield session.measure_throughput()
+    release_memory()
+
+def habitat_predict(session):
+    print("analysis: running deepview_predict()")
+    yield session.habitat_predict()
+    release_memory()
+
+def measure_utilization(session):
+    print("analysis: running measure_utilization()")
+    yield session.measure_utilization()
+    release_memory()
+
+def energy_compute(session):
+    print("analysis: running energy_compute()")
+    yield session.energy_compute()
+    release_memory()
+
+def actual_main(args): 
+    from deepview_profile.analysis.session import AnalysisSession
+    from deepview_profile.exceptions import AnalysisError
+
+    if os.path.exists(args.output):
+        print(
+            "ERROR: The specified output file already exists.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    try: 
+        project_root = os.getcwd()
+        data = { 
+            "analysisState": {
+                "message_type": "analysis",
+                "project_root": project_root,
+                "project_entry_point": args.entry_point,
+                "hardware_info": {},
+                "throughput": {},
+                "breakdown": {},
+                "habitat": {},
+                "additionalProviders": "",
+                "energy": {},
+                "utilization": {}
+            }
+        }
+
+        session = AnalysisSession.new_from(project_root, args.entry_point) 
+        release_memory()
+        message = None
+
+        is_return_all = args.all
+

```

Comment:
Can you add these lines?

data['analysisState']['hardware_info']['hostname'] = platform.node()
        data['analysisState']['hardware_info']['os'] = " ".join(list(platform.uname()))
        data['analysisState']['hardware_info']['gpus'] = []
        
        pynvml.nvmlInit()
        for i in range(pynvml.nvmlDeviceGetCount()):
            handle = pynvml.nvmlDeviceGetHandleByIndex(i)
            device_name = pynvml.nvmlDeviceGetName(handle).decode("utf-8")
            data['analysisState']['hardware_info']['gpus'].append(device_name)
        pynvml.nvmlShutdown()

you will need to import:
import pynvml
import platform

This information helps us show the user what GPU they are currently using.

Reply:
Thank you.
Please, check it in the latest version, it seems to be working.